bluefog/torch/optimizers.py

```diff
@@ -137,7 +137,7 @@ def _timeline_forward_end_hook(module, *unused):
 
 
 class _DistributedOptimizer(torch.optim.Optimizer):
-    def __init__(self, params, model):
+    def __init__(self, params, model, num_steps_per_communication=1):
```

Comment:
Would you please change this back to the exact same style as Horovod?
Because we should provide two allreduce optimizers.  Following is my argument

1) like the horovod style (i.e. this one), which applied the allreduce on the gradient. In this case, it cannot have multiple local updates then communication. So  this num_steps_per_communication should change back to horovod style.

2) We can also do the same allreduce on parameter like the one we did in neighbor allreduce). In that case, num_steps_per_communication should be the same as you did in neighbor allreduce. (I am thinking to add a switch parameter to control use neighbor allreduce or allreduce in forward hook)

Reply:
Done.

Reply:
Done.